extensions/Worker.Extensions.ServiceBus/src/ServiceBusSessionMessageActionsConverter.cs

```diff
@@ -49,7 +54,17 @@ public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
                     throw new InvalidOperationException("Expecting SessionLockedUntil within binding data of session actions and value was not present.");
                 }
 
-                var sessionActionResult = new ServiceBusSessionMessageActions(_settlement, sessionId!.ToString(), sessionLockedUntil.GetDateTimeOffset());
+                // Logic for if isBatched is true, then sessionIdArray will be used to get the sessionId.
+                var sessionIdRepeatedFieldArray = sessionIdArray as RepeatedField<string>;
+
+                if (foundSessionIdArray && (sessionIdRepeatedFieldArray == null || sessionIdRepeatedFieldArray.Count != 1))
+                {
+                     throw new InvalidOperationException($"Expecting batched SessionId within binding data and value was not present. Sessions must be enabled when binding to {nameof(ServiceBusSessionMessageActions)}.");
+                }
+
+                // If sessionIdRepeatedFieldArray has a value, it will only have one value within the array.
```

Comment:
I don't think this is accurate. It looks like [ServiceBus ext](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/ServiceBusTriggerBindingStrategy.cs#L159) will send a value per message in the batch. In which case I think we need to support binding to an array.

Reply:
Hmm I had asked Josh about this before and he said that

>  I think we technically allowing binding to an array of sessionIds but it should only ever have one value.

Let me confirm with him

Reply:
I'll need some clarification for sure. I am confused because `sessionIdArray` will definitely have multiple values. It doesn't seem right for this binding to only consider the first element in that array.

Reply:
Ok so I talked to Josh and he said that:
> Messages and sessions are different concepts. There could be multiple messages but each message would belong to the same session.

So even though we can technically have multiple session ids[ here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Triggers/ServiceBusTriggerBindingStrategy.cs#L176) in the `ServiceBusTriggerBindingStrategy`, we are guaranteed that the value will always be the same. There is no scenario in which we would have different session IDs for the multiple messages.

Reply:
This is really an implementation detail that could theoretically change. Is there a reason that this is important? I would expect the behavior on the isolated worker to match the webjobs extension where you can still bind to an array of session IDs (although it will always have the same value in the current implementation).

Reply:
Looked at the existing implementation that takes a single sessionID in the internal ctor for ServiceBusSessionMessageActions. I think grabbing the first element of the array is fine here given that this is also an internal implementation detail that will need to stay in sync with the webjobs extensions implementation. We should just document in a comment that when using IsBatched=true, all messages will belong to the same session so we can just grab the first element.

Reply:
I updated the comment!

Reply:
Thank you for the clarification @aishwaryabh and @JoshLove-msft. This sounds good to me. Documenting that all values in `sessionIdArray` will always be identical will be helpful. Maybe we can even stop using that property altogether and send only `sessionId` from the WebJobs extension?

I also want to concur that if we go with this implementation which asserts only ever 1 session ID per batch, it will be a breaking change to have multiple session IDs in the future. So we need to be confident that we have no need for that to change, and if it is needed it will be a challenge to change it.

Reply:
> Maybe we can even stop using that property altogether and send only `sessionId` from the WebJobs extension?

I think this would be a breaking change, wouldn't it? Customers may be binding to sessionId as an array.

> I also want to concur that if we go with this implementation which asserts only ever 1 session ID per batch, it will be a breaking change to have multiple session IDs in the future.

I don't think it will happen given the existing public surface area and the implementation details of the Service Bus library, but if we did make this change, we would likely need to EditorBrowsableNever some properties.

Reply:
> Customers may be binding to sessionId as an array.

That was never correctly implemented, so won't be removing any functionality from the customer.

Reply:
Yes, they can do this today so we wouldn't want to break that scenario.